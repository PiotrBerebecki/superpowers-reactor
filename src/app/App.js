import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import theme from './../styled/theme';
import Header from './Header';
import Main from './Main';
import NotFound from './NotFound';
import Puzzles from './../views/Puzzles/index.js';
import otherPuzzleCategories from './../puzzleData/otherPuzzleCategories';

function App() {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <div>
          <Header />
          <Route
            exact
            path="/"
            render={() => <Main><Puzzles category="popular" /></Main>}
          />
          <Route
            path="/:category"
            render={({ match }) => {
              // const otherCategories = ['latest', 'favourite'];
              if (otherPuzzleCategories.includes(match.params.category)) {
                return (
                  <Main><Puzzles category={match.params.category} /></Main>
                );
              } else {
                return <Main><NotFound /></Main>;
              }
            }}
          />
        </div>
      </ThemeProvider>
    </Router>
  );
}

export default App;
