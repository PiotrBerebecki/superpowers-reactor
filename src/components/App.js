import React, { Component } from 'react';

import './App.css';

class App extends Component {
  render() {
    return <div>Superpowers Reactor</div>;
  }
}

export default App;
