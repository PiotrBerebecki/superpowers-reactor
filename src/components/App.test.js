import React from 'react';
import renderer from 'react-test-renderer';

import App from './App.js';

describe('App', () => {
  it('Runs test suite', () => {
    expect(2).toEqual(2);
  });

  // it('renders component correctly', () => {
  //   const component = renderer.create(<App />);
  //
  //   let tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
