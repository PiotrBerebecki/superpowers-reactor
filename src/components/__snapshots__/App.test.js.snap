// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  Superpowers Reactor
</div>
`;
